openmdao/vectors/default_vector.py

```diff
@@ -426,6 +426,26 @@ def set_const(self, val):
         for data in itervalues(self._data):
             data[:] = val
 
+    def dot(self, vec):
+        """
+        Compute the dot product of the real parts of the current vec and the incoming vec.
+
+        Parameters
+        ----------
+        vec : <Vector>
+            the vector whose values self is set to.
+
+        Returns
+        -------
+        float
+            The computed dot product value.
+        """
+        global_sum = 0
+        for set_name, data in iteritems(self._data):
+            global_sum += np.sum(self._data[set_name] * vec._data[set_name])
```

Comment:
This will create a temporary array of the same size as the data. Why not use `np.dot` or `np.einsum`?

Reply:
Corrections have been made with hwangjt's inputs. np.sum has been replaced with np.dot. Thanks.